packages/kit/src/runtime/server/cookie.js

```diff
@@ -59,13 +74,19 @@ export function get_cookies(request, url) {
 		 * @param {import('cookie').CookieParseOptions} [opts]
 		 */
 		get(name, opts) {
-			const c = new_cookies[name];
-			if (
-				c &&
-				domain_matches(url.hostname, c.options.domain) &&
-				path_matches(url.pathname, c.options.path)
-			) {
-				return c.value;
+			// Look for the most specific matching cookie from new_cookies
+			const best_match = Array.from(new_cookies.values())
+				.filter((c) => {
+					return (
+						c.name === name &&
+						domain_matches(url.hostname, c.options.domain) &&
+						path_matches(url.pathname, c.options.path)
+					);
+				})
+				.sort((a, b) => b.options.path.length - a.options.path.length)[0];
```

Comment:
I'm guessing that sorting by length works because browsers don't send cookies for unrelated paths, but it won't work after the server creates a cookie for an unrelated path. Is this the remaining bug you mentioned @Rich-Harris ?

Reply:
Ah no. An unrelated path would've already been filtered out in the previous step.